src/macaron/repo_finder/repo_finder.py

```diff
@@ -148,3 +131,74 @@ def to_repo_path(purl: PackageURL, available_domains: list[str]) -> str | None:
             fragment="",
         )
     )
+
+
+def find_source(purl_string: str, input_repo: str | None) -> bool:
+    """Perform repo and commit finding for a passed PURL, or commit finding for a passed PURL and repo.
+
+    Parameters
+    ----------
+    purl_string: str
+        The PURL string of the target.
+    input_repo: str | None
+        The repository path optionally provided by the user.
+
+    Returns
+    -------
+    bool
+        True if the source was found.
+    """
+    try:
+        purl = PackageURL.from_string(purl_string)
+    except ValueError as error:
+        logger.error("Could not parse PURL: %s", error)
+        return False
+
+    found_repo = input_repo
+    if not input_repo:
+        logger.debug("Searching for repo of PURL: %s", purl)
+        found_repo = find_repo(purl)
+
+    if not found_repo:
+        logger.error("Could not find repo for PURL: %s", purl)
+        return False
+
+    # Disable other loggers for cleaner output.
+    analyzer_logger = logging.getLogger("macaron.slsa_analyzer.analyzer")
+    analyzer_logger.disabled = True
+    git_logger = logging.getLogger("macaron.slsa_analyzer.git_url")
+    git_logger.disabled = True
+
+    # Prepare the repo.
+    logger.debug("Preparing repo: %s", found_repo)
+    repo_dir = os.path.join(global_config.output_path, GIT_REPOS_DIR)
+    git_obj = prepare_repo(
```

Comment:
Instead of cloning the repo, I wonder if we can use the GitHub API to get the tags?
```
curl -s https://api.github.com/repos/npm/node-semver/tags | jq '.[] | {tag: .name, commit: .commit.sha}'
```